cornflow/schemas/schema_manager.py

```diff
@@ -0,0 +1,368 @@
+        In jsonschema arrays are similar to python lists.
+        The object in jsonschema is in the following format:
+        "object_name": {"type":"array", "items":{format_of_items}}
+
+        The schema_dict object use the format:
+        {"schema_name": [{"name":"field1", "type": "field1_type", "many": False, "required":(True or False)
+
+        :param item: The jsonschema item (key, value)
+        The format of the item is: ("object_name", {"type":"object", "properties":{"a": {...}, "b": {...}})
+
+        :return: The schema name
+        """
+        name = item[0]
+        content = item[1]["items"]
+        schema_name = self._get_schema_name(name)
+        if "type" in content and content["type"] == "object":
+            self.schema_dict.update(
+                {schema_name: [self._get_element_dict(i, self._get_required(content))
+                               for i in content["properties"].items()]}
+            )
+        elif "$ref" in content:
+            schema_name = self._get_ref((None, content))
+        else:
+            self.schema_dict.update(
+                {schema_name: [self._get_element_dict(i, self._get_required(content)) for i in content.items()]})
+        return schema_name
+       
+    def _get_field_dict(self, item, required_list=None):
+        """
+        Transform a "normal" item from the jsonschema in a dict for the schema_dict and return it.
+        This is used for items that will directly translate into fields.
+        
+        :param item: The jsonschema item in format (key, value)
+        :param required_list: a list of the fields required in the parent object.
+        
+        :return: the schema_dict for this item
+        """
+        
+        d = dict(
+            name=item[0],
+            type=self._get_type(item[1]["type"]),
+            required=(item[0] in required_list),
+            allow_none=("null" in item[1]["type"]),
+            many=False)
+        return d
+    
+    def _get_ref(self, item):
+        """
+        Get the name of the schema for a jsonschema reference.
+        jsonschema definitions are parsed first and corresponign schema are created so a schema should exist
+         corresponding to the reference.
+        
+        :param item: The jsonschema item in format (key, value)
+        The value should be in the following format: {"$ref": "#/definitions/object_name"}
+        
+        :return The schema name (_get_schema_name(object_name))
+        """
+        content = item[1]
+        ref = re.search("definitions/(.+)", content["$ref"]).group(1)
+        return self._get_schema_name(ref)
+    
+    def _get_type(self, json_type):
+        """
+        Translate the type between jsonschema and schema_dict.
+        
+        :param json_type: the type in jsonschema
+        
+        :return: the type in schema_dict.
+        """
+        if type(json_type) is list:
```

Comment:
I think it's better to use `isinstance()`. It gets also the subclasses of list.

Reply:
I think in general it is a good idea to do that. But here, I do not see how it could be a subclass (it is read from json) and why we should accept a subclass if it is.